Sources/TuistLoader/Utils/PluginsHelper.swift

```diff
@@ -0,0 +1,55 @@
+import Foundation
+import TSCBasic
+import TuistCore
+import TuistGraph
+import TuistSupport
+
+protocol PluginsHelping {
+    func templatePath(
+        for pluginName: String,
+        resourceName: String,
+        resourceSynthesizerPlugins: [ResourceSynthesizerPlugin]
+    ) throws -> AbsolutePath
+}
+
+enum PluginsHelperError: FatalError, Equatable {
+    case pluginNotFound(String, [String])
+    case resourceTemplateNotFound(name: String, plugin: String)
+
+    var type: ErrorType {
+        switch self {
+        case .pluginNotFound,
+             .resourceTemplateNotFound:
+            return .abort
+        }
+    }
+
+    var description: String {
+        switch self {
+        case let .pluginNotFound(name, availablePlugins):
+            return "Plugin \(name) was not found. Available plugins: \(availablePlugins.joined(separator: ", "))"
+        case let .resourceTemplateNotFound(name: name, plugin: pluginName):
+            return "No template \(name) found in a plugin \(pluginName)"
+        }
+    }
+}
+
+final class PluginsHelper: PluginsHelping {
```

Comment:
Thoughts on this name being confused with `PluginService`, it looks like for the most part this helper deals with loading template paths for a given plugin.

Wonder if something more verbose like `PluginsTemplatePathHelper` is easier to understand at a glance? Let me know though what you think!

Reply:
sure, I'll rename it!